In [ ]:
from google.colab import drive
drive.mount('/content/gdrive')

Mounted at /content/gdrive


In [ ]:
cd /content/gdrive/My Drive/Notebooks

/content/gdrive/My Drive/Notebooks


"Hello everyone, in this video, we'll be exploring the Iris dataset and building a logistic regression model to classify iris species. We'll start by loading the dataset and performing exploratory data analysis. Then, we'll preprocess the data, split it into training and testing sets, and train our logistic regression model. Finally, we'll evaluate the model's performance and make predictions on new data. Let's dive in!"

importing necessary Python libraries for data analysis and visualization.

In [ ]:
from sklearn import datasets
import pandas as pd #facilitates data manipulation.
import numpy as np # numpy handles numerical computations.
import matplotlib.pyplot as plt # enables plotting of data.

This line loads the Iris dataset from scikit-learn's datasets module.

In [ ]:
iris=datasets.load_iris()

This line extracts the first two features (columns) from the Iris dataset and assigns them to the variable X. It selects all rows of the dataset but only the first two columns, allowing for subsequent analysis or visualization focusing on these specific features.

In [ ]:
X = iris.data[:, :2]

This line assigns the target variable (species labels) from the Iris dataset to the variable y.

In [ ]:
y = iris.target

Typically represents a dataset object.

In [ ]:
type(iris)

sklearn.utils._bunch.Bunch

Printing iris displays the information about the Iris dataset, including its features, target variable, and other metadata.

In [ ]:
print(iris)

{'data': array([[5.1, 3.5, 1.4, 0.2],
       [4.9, 3. , 1.4, 0.2],
       [4.7, 3.2, 1.3, 0.2],
       [4.6, 3.1, 1.5, 0.2],
       [5. , 3.6, 1.4, 0.2],
       [5.4, 3.9, 1.7, 0.4],
       [4.6, 3.4, 1.4, 0.3],
       [5. , 3.4, 1.5, 0.2],
       [4.4, 2.9, 1.4, 0.2],
       [4.9, 3.1, 1.5, 0.1],
       [5.4, 3.7, 1.5, 0.2],
       [4.8, 3.4, 1.6, 0.2],
       [4.8, 3. , 1.4, 0.1],
       [4.3, 3. , 1.1, 0.1],
       [5.8, 4. , 1.2, 0.2],
       [5.7, 4.4, 1.5, 0.4],
       [5.4, 3.9, 1.3, 0.4],
       [5.1, 3.5, 1.4, 0.3],
       [5.7, 3.8, 1.7, 0.3],
       [5.1, 3.8, 1.5, 0.3],
       [5.4, 3.4, 1.7, 0.2],
       [5.1, 3.7, 1.5, 0.4],
       [4.6, 3.6, 1. , 0.2],
       [5.1, 3.3, 1.7, 0.5],
       [4.8, 3.4, 1.9, 0.2],
       [5. , 3. , 1.6, 0.2],
       [5. , 3.4, 1.6, 0.4],
       [5.2, 3.5, 1.5, 0.2],
       [5.2, 3.4, 1.4, 0.2],
       [4.7, 3.2, 1.6, 0.2],
       [4.8, 3.1, 1.6, 0.2],
       [5.4, 3.4, 1.5, 0.4],
       [5.2, 4.1, 1.5, 0.1],
       [5.5, 4.2, 1.4, 0.2],
     

providing access to its attributes

In [ ]:
iris.keys()

dict_keys(['data', 'target', 'frame', 'target_names', 'DESCR', 'feature_names', 'filename', 'data_module'])

This command displays the first five rows of the data contained within the Iris dataset, providing a glimpse into the feature values of the dataset.

In [ ]:
iris.data[:5]

array([[5.1, 3.5, 1.4, 0.2],
       [4.9, 3. , 1.4, 0.2],
       [4.7, 3.2, 1.3, 0.2],
       [4.6, 3.1, 1.5, 0.2],
       [5. , 3.6, 1.4, 0.2]])

Features names is showing the column names in IRIS dataset

In [ ]:
iris.feature_names

['sepal length (cm)',
 'sepal width (cm)',
 'petal length (cm)',
 'petal width (cm)']

retrieves the target values of the first five samples in the Iris dataset.

In [ ]:
iris.target[:5]

array([0, 0, 0, 0, 0])

'setosa', 'versicolor', 'virginica' is the target names of the dataset.

In [ ]:
iris.target_names

array(['setosa', 'versicolor', 'virginica'], dtype='<U10')

the description of the Iris dataset, providing detailed information about its features, target variable, and dataset source.

In [ ]:
print(iris.DESCR)

.. _iris_dataset:

Iris plants dataset
--------------------

**Data Set Characteristics:**

    :Number of Instances: 150 (50 in each of three classes)
    :Number of Attributes: 4 numeric, predictive attributes and the class
    :Attribute Information:
        - sepal length in cm
        - sepal width in cm
        - petal length in cm
        - petal width in cm
        - class:
                - Iris-Setosa
                - Iris-Versicolour
                - Iris-Virginica
                
    :Summary Statistics:

    ============== ==== ==== ======= ===== ====================
                    Min  Max   Mean    SD   Class Correlation
    ============== ==== ==== ======= ===== ====================
    sepal length:   4.3  7.9   5.84   0.83    0.7826
    sepal width:    2.0  4.4   3.05   0.43   -0.4194
    petal length:   1.0  6.9   3.76   1.76    0.9490  (high!)
    petal width:    0.1  2.5   1.20   0.76    0.9565  (high!)
    ============== ==== ==== ======= ===== ===========

 creating a DataFrame df from the Iris dataset's features, with column names set to the feature names.

In [ ]:
df=pd.DataFrame(iris.data,columns=iris.feature_names)

Printing the first 5 rows of the DataFrame using.

In [ ]:
df.head()

sepal length (cm)  sepal width (cm)  petal length (cm)  petal width (cm)
0                5.1               3.5                1.4               0.2
1                4.9               3.0                1.4               0.2
2                4.7               3.2                1.3               0.2
3                4.6               3.1                1.5               0.2
4                5.0               3.6                1.4               0.2

It displays the 150 columns and 4 rows is there.

In [ ]:
df.shape

(150, 4)

Adding the target variable to the DataFrame as a new column named "target".

In [ ]:
df["target"]=pd.Series(iris.target)

Once again the printing the first 5 rows of the DataFrame using. And here we see the new coulmn is add to dataset.

In [ ]:
df.head()

sepal length (cm)  sepal width (cm)  petal length (cm)  petal width (cm)  \
0                5.1               3.5                1.4               0.2   
1                4.9               3.0                1.4               0.2   
2                4.7               3.2                1.3               0.2   
3                4.6               3.1                1.5               0.2   
4                5.0               3.6                1.4               0.2   

   target  
0       0  
1       0  
2       0  
3       0  
4       0

Printing information about the DataFrame using

In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 150 entries, 0 to 149
Data columns (total 5 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   sepal length (cm)  150 non-null    float64
 1   sepal width (cm)   150 non-null    float64
 2   petal length (cm)  150 non-null    float64
 3   petal width (cm)   150 non-null    float64
 4   target             150 non-null    int64  
dtypes: float64(4), int64(1)
memory usage: 6.0 KB


Generates descriptive statistics of the DataFrame using

In [ ]:
df.describe()

sepal length (cm)  sepal width (cm)  petal length (cm)  \
count         150.000000        150.000000         150.000000   
mean            5.843333          3.057333           3.758000   
std             0.828066          0.435866           1.765298   
min             4.300000          2.000000           1.000000   
25%             5.100000          2.800000           1.600000   
50%             5.800000          3.000000           4.350000   
75%             6.400000          3.300000           5.100000   
max             7.900000          4.400000           6.900000   

       petal width (cm)      target  
count        150.000000  150.000000  
mean           1.199333    1.000000  
std            0.762238    0.819232  
min            0.100000    0.000000  
25%            0.300000    0.000000  
50%            1.300000    1.000000  
75%            1.800000    2.000000  
max            2.500000    2.000000

Groups the DataFrame by target and displays the size of each group.

In [ ]:
df.groupby('target').size()

target
0    50
1    50
2    50
dtype: int64

This code iterates over the target values in the Iris dataset, mapping each value to its corresponding species name ("setosa", "versicolor", or "virginica"). It then adds a new column "species" to the DataFrame df, containing the mapped species names for each target value.

In [ ]:
species=[]
for i in range(len(iris['target'])):
  if iris['target'][i]==0:
    species.append("setosa")
  elif iris['target'][i]==1:
    species.append("versicolor")
  else:
    species.append("virginica")

df['species']=species

Once again the printing the first 5 rows of the DataFrame using. And here we see the new coulmn is add to dataset.

In [ ]:
df.head()

sepal length (cm)  sepal width (cm)  petal length (cm)  petal width (cm)  \
0                5.1               3.5                1.4               0.2   
1                4.9               3.0                1.4               0.2   
2                4.7               3.2                1.3               0.2   
3                4.6               3.1                1.5               0.2   
4                5.0               3.6                1.4               0.2   

   target species  
0       0  setosa  
1       0  setosa  
2       0  setosa  
3       0  setosa  
4       0  setosa

Groups the DataFrame by species and displays the size of each group.

In [ ]:
df.groupby('species').size()

species
setosa        50
versicolor    50
virginica     50
dtype: int64

Checking for the missing values in the DataFrame using

In [ ]:
df.isnull().sum()

sepal length (cm)    0
sepal width (cm)     0
petal length (cm)    0
petal width (cm)     0
target               0
species              0
dtype: int64

Printing the column names in the Dataset. And also checking the datatype and its a object type data.

In [ ]:
df.columns

Index(['sepal length (cm)', 'sepal width (cm)', 'petal length (cm)',
       'petal width (cm)', 'target', 'species'],
      dtype='object')

This code imports the train_test_split function from sklearn.model_selection, which is used to split data into training and testing sets. It then creates a feature matrix X by dropping the columns 'target' and 'species' from the DataFrame df, and prints the data type of X, which is a pandas DataFrame.

In [ ]:
from sklearn.model_selection import train_test_split
X=df.drop(['target','species'],axis=1)
print(type(X))

<class 'pandas.core.frame.DataFrame'>


This code converts the DataFrame X to a NumPy array using the to_numpy() method and selects only the columns at index 2 and 3 (0-based indexing) from the array, effectively reducing the feature matrix X to contain only those two selected columns.

In [ ]:
X=X.to_numpy()[:,(2,3)]

This code prints the first five rows of the feature matrix X, displaying the selected subset of features for each sample in the dataset.

In [ ]:
X[:5]

array([[1.4, 0.2],
       [1.4, 0.2],
       [1.3, 0.2],
       [1.5, 0.2],
       [1.4, 0.2]])

This code returns the data type of the variable X, indicating whether it is a DataFrame or a NumPy array.

In [ ]:
type(X)

numpy.ndarray

This code assigns the target variable from the Iris dataset to the variable y, representing the species labels for each sample in the dataset.

In [ ]:
y=iris['target']

This line of code splits the dataset into training and testing sets, with 80% of the data used for training (X_train, y_train) and 20% for testing (X_test, y_test). The random_state parameter ensures reproducibility of the split, maintaining consistency across multiple runs.

In [ ]:
X_train,X_test,y_train,y_test=train_test_split(X,y,test_size=0.2,random_state=12)

Imports Logistic Regression model from scikit-learn.Trains the Logistic Regression model on the training data.Fit the model.

In [ ]:
from sklearn.linear_model import LogisticRegression
log_reg=LogisticRegression()
log_reg.fit(X_train,y_train)

LogisticRegression()

Makes predictions on the test data.This code uses the trained logistic regression model log_reg to make predictions on the test data X_test, assigning the predicted species labels to the variable test_prediction.

In [ ]:
test_prediction=log_reg.predict(X_test)

The variable test_prediction contains the predicted species labels for the test data, as generated by the logistic regression model.

In [ ]:
test_prediction

array([0, 2, 0, 1, 2, 2, 2, 0, 2, 0, 1, 0, 0, 0, 1, 2, 2, 1, 0, 1, 0, 1,
       2, 1, 0, 2, 2, 1, 0, 0])

This code prints the precision, recall, and confusion matrix metrics for evaluating the performance of the classification model on the test data. It utilizes the classification_report function from the metrics module in scikit-learn to generate a detailed report.

In [ ]:
from sklearn import metrics
print("precision,Recall,Confusion matrix")
print(metrics.classification_report(y_test,test_prediction,digits=3))

precision,Recall,Confusion matrix
              precision    recall  f1-score   support

           0      1.000     1.000     1.000        12
           1      1.000     0.889     0.941         9
           2      0.900     1.000     0.947         9

    accuracy                          0.967        30
   macro avg      0.967     0.963     0.963        30
weighted avg      0.970     0.967     0.967        30



Confusion matrix to analyze the classification performance by comparing the actual and predicted labels.

In [ ]:
metrics.confusion_matrix(y_test,test_prediction)

array([[12,  0,  0],
       [ 0,  8,  1],
       [ 0,  0,  9]])

This code prepares a new data point xpredict with feature values (3.4, 5.6), reshaping it to match the model's input shape. It then uses the trained logistic regression model log_reg to predict the class of the new data point and prints the predicted class. If the predicted class is 2, it is labeled as "Versicolor".

In [ ]:
xpredict=np.array([3.4,5.6])
xpredict.shape=(1,2)
print(xpredict)
test_prediction=log_reg.predict(xpredict)
if test_prediction==2:
  test_prediction="Versicolor"
print("this data belongs to class",test_prediction)

[[3.4 5.6]]
this data belongs to class Versicolor
